backend/src/main/java/com/google/step/utils/AdvertiserUtil.java

```diff
@@ -15,62 +15,107 @@
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
-import com.google.appengine.api.users.User;
 import com.google.appengine.api.datastore.Entity;
-import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.EntityNotFoundException;
 import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.users.User;
 
 
 /**
  * This class is designed to be the single point of responsibility for storing and reading
- * Advertiser entities from the datastore.
- * Advertiser refers to the entity in the datastore, representing a User provided by Google
- * Authentication using our application.
+ * Advertiser entities from the datastore. Advertiser refers to the entity in the datastore,
+ * representing a User provided by Google Authentication using our application.
  */
 public final class AdvertiserUtil {
 
-    /**
-     * Checks whether the user object passed in exists in datastore as an advertiser
-     * @param user the google user to be checked
-     * @return     true if the user is contained in the datastore as an advertiser, false otherwise
-     * @throws DatastoreFailureException if a datastore error occurs
-     */
-    public static boolean advertiserExistsInDatastore(User user) {
-        try {
-            DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-            Key advertiserKey = createAdvertiserKey(user);
-            datastore.get(advertiserKey);
-            return true;
-        } catch (EntityNotFoundException e) {
-            return false;
-        } finally {
-            return false;
-        }
-    }
+  public static final String ADVERTISER_ENTITY_NAME = "Advertiser";
 
-    /**
-     * Puts the user object passed in to the datastore as an advertiser entity
-     * @param user the google user to be added
-     * @throws java.util.ConcurrentModificationException if the entity group that the user entity
-     *  belongs to was being modified concurrently
-     * @throws DatastoreFailureException if any other datastore error occurs
-     */
-    public static void putAdvertiserInDatastore(User user) {
-        Key advertiserKey = createAdvertiserKey(user);
-        Entity userEntity = new Entity(advertiserKey);
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        datastore.put(userEntity);
+  /**
+   * Checks whether the user object passed in exists in datastore as an advertiser
+   *
+   * @param user the google user to be checked
+   * @return true if the user is contained in the datastore as an advertiser, false otherwise
+   * @throws DatastoreFailureException if a datastore error occurs
+   */
+  public static boolean advertiserExistsInDatastore(User user) {
+    try {
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      Key advertiserKey = createAdvertiserKey(user);
```

Comment:
which part throws the EntityNotFoundException? Maybe have that be the only part of the try block.

Reply:
Moved the datastore and advertiserKey lines out of the try block